src/transformStateWithClones.js

```diff
@@ -7,7 +7,33 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'clear':
+        currentState = {};
+        break;
+
+      case 'addProperties':
+        if (typeof action.extraData === 'object' && action.extraData !== null) {
+          Object.assign(currentState, action.extraData);
+        }
```

Comment:
Consider using the spread operator to merge objects as it is more modern and concise than Object.assign. For example, you can use `currentState = { ...currentState, ...action.extraData };`.